ArduPlane/soaring.cpp

```diff
@@ -47,175 +47,134 @@ void Plane::update_soaring() {
     }
 
     switch (control_mode->mode_number()) {
-    default:
-        // nothing to do
-        break;
-    case Mode::Number::AUTO:
-    case Mode::Number::FLY_BY_WIRE_B:
-    case Mode::Number::CRUISE:
-        // Test for switch into thermalling mode
-        g2.soaring_controller.update_cruising();
-
-        if (g2.soaring_controller.check_thermal_criteria()) {
-            gcs().send_text(MAV_SEVERITY_INFO, "Soaring: Thermal detected, entering %s", mode_loiter.name());
-            set_mode(mode_loiter, ModeReason::SOARING_THERMAL_DETECTED);
-        }
-        break;
+        default:
```

Comment:
Removing this `default:` would encourage someone creating a mode to consider what has to happen in here.